app/Controllers/Http/BooksController.ts

```diff
@@ -15,21 +18,35 @@ export default class BooksController {
   }
 
   public async store({ request, response }: HttpContextContract) {
-    const book = request.body() as BookDTO
-    const books = this.booksRepository.add(book)
-    response.status(201).json(books)
+    try {
+      const params = await request.validate({ schema: BookSchema })
+      const parsedParams = new BookParser(params).parse()
```

Comment:
Este trecho acaba se repetindo em partes (eu n entendi pq o `const params` do `update` é diferente) com o método `update`. Além disso essas duas linhas tem uma responsabilidade unica que é filtrar parametros válidos. Talvez valha extrair essas 2 linhas para um método privado nesta classe, algo como:

```suggestion
      const params = await permitedParams(request.body())
```

Reply:
Criei um método privado pra fazer o parser, assim recebe direto o request, valida este request e, em caso de sucesso, devolve os dados parseados.